src/main/java/org/jbpm/bootstrap/service/controllers/IndexController.java

```diff
@@ -192,24 +203,53 @@ protected String resolveApplicationType(Project project) {
         }
     }
 
-    protected void waitForGeneratedProject(File generatedProject) {
-        try {
-            long totalWaitTime = 0;
-            while (true) {
+    protected void waitForGeneratedProject(File generatedProject, long processInstanceId) throws Exception {
+        // if we had errors dont even start waiting
+        if(haveProcessInstanceErrors(processInstanceId)) {
```

Comment:
maybe change the error check to throw directly error if found so we then have the error details already extracted?